### DISCUSSION TITLE
Initial relative permittivity distribution for adjoint optimization?

### DISCUSSION THREAD

**Discussion by satadrudas** [NanoComp/meep#2596] [Q&A] [answered]
Hi,
From all the adjoint optimization example in this repository and few papers that i have read on MEEP adjoint optimization, in almost all of the the initial  *rho* value is 0.5 . 

![image](https://github.com/NanoComp/meep/assets/38806771/bb3e4bce-66dd-4bd2-a99b-ac502571415d)

Is there any special reason for this or arbitrarily chosen? Can we expect to see better results when initial *rho* is random [0,1]?

 The reason I ask this is because atleast when I am running an adjoint optimization with random *rho*, the results are not as good as when initialized with 0.5

**Comment by smartalecH** [comment]
This is just a common convention within topology optimization, and im not sure if there's an agreed-upon reason _why_ this is the case. I like to say that starting with a uniform field at 0.5 prevents any bias. But with these kinds of things, it's really hard to draw conclusions... 🤷 

Ultimately things will be problem dependent™️

**Reply by satadrudas** [reply]
got it, thanks

**Comment by stevengj** [comment, marked as answer]
In a problem with lots of local optima, random starting points tend to lead to more irregular-looking structures (often with lots of extraneous structures that don't affect performance much but make the geometry look more complicated) while high-symmetry starting points tend to lead to more regular structures with fewer extraneous features.  The former aren't necessarily worse than the latter in performance, but people tend to like more regular structures if the performance is comparable.

**Reply by satadrudas** [reply]
Hi, thanks for the input. I still have a doubt though. Even when going for a random starting point with more irregular structures, how do we know if they(extraneous structures) don't affect the performance of the device, in a good/bad? What I had in mind was that with different initial points we could hope to get better designs.

**Reply by stevengj** [reply]
We don't know.  Welcome to the exciting world of non-convex optimization.

**Reply by satadrudas** [reply]
excited to be here! :)
